hyppo/independence/mirf.py

```diff
@@ -285,3 +295,27 @@ def test_cutoff(self, x, z, y):
         X_permutedZ = np.hstack((x, permuted_Z))
         null = self.statistic(X_permutedZ, y)
         return observe_stat, null
+
+    def test_diff(self, x, z, y, reps=1000, workers=1):
+        XZ = np.hstack((x, z))
+        observe_stat, observe_pos = self.statistic(XZ, y, return_pos=True)
+
+        # Compute statistic for permuted sets
+        permuted_Z = np.random.permutation(z)
+        X_permutedZ = np.hstack((x, permuted_Z))
+        perm_stat, perm_pos = self.statistic(X_permutedZ, y, return_pos=True)
```

Comment:
Shouldn't this retrain a forest? 

There should be one forest for 

(XZ, y)

And one for (Xzperm, y). Maybe I missed something.

Reply:
`statistic` will retrain the forest in this case.

Reply:
Due to the special case we have regarding forest training (split for each tree, no common test set), keeping the posteriors makes the most sense to me.